### ISSUE TITLE
staged_predict for Boosted Tree classifier

### ISSUE THREAD

**Issue by samanemami** [tensorflow/tensorflow#51834]
<emThe issue related to the performance of the [BoostedTree classifier](https://github.com/tensorflow/estimator/blob/781c0d30c6bf100aa174591dd97cb70fc39d294d/tensorflow_estimator/python/estimator/canned/boosted_trees_test.py#L403) in terms of accuracy.</em>

**System information**

- OS Platform and Distribution: Linux Ubuntu 16.04, Windows 10
- TensorFlow installed from (source or binary): pip install TensorFlow
- TensorFlow version (use command below): 2.4.1
- Python version: 3.6

** Describe the model**
BoostedTree classifier is a model introduced by [N Ponomareva, T Colthurst, G Hendry.2017](https://ieeexplore.ieee.org/abstract/document/8257910).
It is built over the Xgboost idea and learning is done through building one layer of decision tree regressor over N boosting iteration.

**Describe the current behavior**
Return the accuracy of the ensemble model with n_tree

**Describe the expected behavior**
I need to have the accuracy of each individual tree in the ensemble model.
For example, the Gradient Boosting of Sklearn has the [staged_predict method](https://github.com/scikit-learn/scikit-learn/blob/2beed5584/sklearn/ensemble/_gb.py#L1270).
Is there any similar method in tf.BoostedTree as well?

**Standalone code to reproduce the issue**
You may find an example of the model, [here](https://www.tensorflow.org/tutorials/estimator/boosted_trees_model_understanding)

**Comment by samanemami** [comment]
Hi @mihaimaruseac, May I ask for your help on this issue, please?
